common/src/main/java/dev/ftb/mods/ftbchunks/net/ShareWaypointPacket.java

```diff
@@ -0,0 +1,85 @@
+package dev.ftb.mods.ftbchunks.net;
+
+import dev.architectury.networking.NetworkManager;
+import dev.ftb.mods.ftbchunks.api.FTBChunksAPI;
+import dev.ftb.mods.ftblibrary.util.NetworkHelper;
+import dev.ftb.mods.ftbteams.api.FTBTeamsAPI;
+import dev.ftb.mods.ftbteams.api.Team;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.GlobalPos;
+import net.minecraft.core.UUIDUtil;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.chat.*;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.server.players.PlayerList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+public record ShareWaypointPacket(String name, GlobalPos position, ShareType shareType, List<UUID> targets) implements CustomPacketPayload {
+    public static final Type<ShareWaypointPacket> TYPE = new Type<>(FTBChunksAPI.rl("share_waypoint_packet"));
+
+    public static final StreamCodec<FriendlyByteBuf, ShareWaypointPacket> STREAM_CODEC = StreamCodec.composite(
+            ByteBufCodecs.STRING_UTF8, ShareWaypointPacket::name,
+            GlobalPos.STREAM_CODEC, ShareWaypointPacket::position,
+            ShareType.STREAM_CODEC, ShareWaypointPacket::shareType,
+            UUIDUtil.STREAM_CODEC.apply(ByteBufCodecs.list()), ShareWaypointPacket::targets,
+            ShareWaypointPacket::new
+    );
+
+    @Override
+    public Type<ShareWaypointPacket> type() {
+        return TYPE;
+    }
+
+    public static void handle(ShareWaypointPacket message, NetworkManager.PacketContext context) {
+        context.queue(() -> {
+            ServerPlayer serverPlayer = (ServerPlayer) context.getPlayer();
+            PlayerList playerList = serverPlayer.getServer().getPlayerList();
+            ChatType.Bound bound2 = ChatType.bind(ChatType.CHAT, serverPlayer).withTargetName(serverPlayer.getDisplayName());
+            List<ServerPlayer> playersToSend = switch (message.shareType) {
```

Comment:
Not a massive issue but I do tend to prefer some more whitespace around different sections of code. I'd likely have split this list up from the above `bound2` and the below `for`